### DISCUSSION TITLE
jsr publish doesn't warn on "slow types", but JSR score still suffers because of "slow types"

### DISCUSSION THREAD

**Discussion by Thesephi** [jsr-io/jsr#645] [Q&A]
Dear community,

I've been maintaining a JSR lib at 100% JSR score for several months. But in my latest release, it dropped to 76%, due to "slow types":

<img width="1200" alt="jsr-slow-types" src="https://github.com/user-attachments/assets/86dda127-0043-4da5-86fb-ab8654dd8e9d">

My question is: **is there any command we can run to reveal which exact types are being "slow" here?**

I've tried several things based on my knowledge:

```bash
deno publish --dry-run
Checking for slow types in the public API...
Simulating publish of @dklab/oak-routing-ctrl@0.8.2 with files:
   file:///oak-routing-ctrl/CHANGELOG.md (2.91KB)
   file:///oak-routing-ctrl/LICENSE (1.05KB)
   file:///oak-routing-ctrl/README.md (9.41KB)
   file:///oak-routing-ctrl/deps.ts (1.25KB)
   file:///oak-routing-ctrl/jsr.json (354B)
   file:///oak-routing-ctrl/mod.ts (553B)
   file:///oak-routing-ctrl/src/Constants.ts (173B)
   file:///oak-routing-ctrl/src/Controller.ts (998B)
   file:///oak-routing-ctrl/src/ControllerMethodArgs.ts (6.7KB)
   file:///oak-routing-ctrl/src/Delete.ts (901B)
   file:///oak-routing-ctrl/src/Get.ts (883B)
   file:///oak-routing-ctrl/src/Head.ts (889B)
   file:///oak-routing-ctrl/src/Options.ts (907B)
   file:///oak-routing-ctrl/src/Patch.ts (895B)
   file:///oak-routing-ctrl/src/Post.ts (889B)
   file:///oak-routing-ctrl/src/Put.ts (883B)
   file:///oak-routing-ctrl/src/Store.ts (688B)
   file:///oak-routing-ctrl/src/oasStore.ts (2.32KB)
   file:///oak-routing-ctrl/src/useOakServer.ts (1.62KB)
   file:///oak-routing-ctrl/src/useOas.ts (2.43KB)
   file:///oak-routing-ctrl/src/utils/getUserSuppliedDecoratedMethodName.ts (807B)
   file:///oak-routing-ctrl/src/utils/inspect.ts (145B)
   file:///oak-routing-ctrl/src/utils/logger.ts (367B)
   file:///oak-routing-ctrl/test_utils/mockRequestInternals.ts (1.77KB)
Warning Aborting due to --dry-run
```
gives me a clean output (success)

Event GitHub action completes <ins>successfully</ins>: https://github.com/Thesephi/oak-routing-ctrl/actions/runs/9898400859/job/27345131692#step:7:9

```bash
deno check --all mod.ts 
Check file:///oak-routing-ctrl/mod.ts
```

also gives me a clean output. No warning, no error...

---

Even I can guesstimate what could possibly have been judged as "slow types", it's a best-effort endeavour at best, and likely a yak-shaving process that sees no easy end.

To this end, I'd love to ask here, if anyone is aware of a method to "print out all slow types"? If there's no such thing yet, may I get pointed to the person or team who's most likely working on a similar topic, so that I might offer some assistance (if it's in my capacity)?

Last but not least: if anyone needs a live example to reproduce this, please feel free to clone the repo at this tag: `https://github.com/Thesephi/oak-routing-ctrl/tree/v0.8.2` then try `deno publish --dry-run` on it!

Thank you in advance for your insights on this matter!

**Comment by sergey-shandar** [comment]
Same here: 
- JSR: https://jsr.io/@functionalscript/functionalscript@0.2.0, `slow-types: 0/5`
- Publishing: https://github.com/functionalscript/functionalscript/actions/runs/12289229731/job/34294400355#step:10:66
